apps/condo/domains/miniapp/schema/B2BAppAccessRightSet.js

```diff
@@ -35,15 +75,112 @@ const B2BAppAccessRightSet = new GQLListSchema('B2BAppAccessRightSet', {
+                    app: { id: nextItem.app },
+                    deletedAt: null,
+                })
+                
+                const globalRightSet = rightSetsForApp.find(set => set.type === ACCESS_RIGHT_SET_GLOBAL_TYPE)
+                if (!globalRightSet) {
+                    throw new GQLError(ERRORS.GLOBAL_RIGHT_SET_REQUIRED, context)
+                }
+                const sameTypeRightSets = rightSetsForApp.filter(set => set.type === nextItem.type)
+                if (sameTypeRightSets.length >= ACCESS_RIGHT_SET_MAX_ITEMS_FOR_TYPE[nextItem.type]) {
+                    throw new GQLError({
+                        ...ERRORS.TOO_MANY_ITEMS_OF_TYPE,
+                        messageInterpolation: { type: nextItem.type, maximum: ACCESS_RIGHT_SET_MAX_ITEMS_FOR_TYPE[nextItem.type] },
+                    }, context)
+                }
+
+                if (nextItem.type !== ACCESS_RIGHT_SET_GLOBAL_TYPE) {
+                    const permissionDiff = getPermissionsDiff(globalRightSet, nextItem)
+                    const enabledPermissionsDiff = getEnabledPermissions(permissionDiff)
+                    if (Object.keys(enabledPermissionsDiff).length) {
+                        throw new GQLError(ERRORS.TOO_MANY_PERMISSIONS, context)
+                    }
+                }
+            }
+        },
+        async afterChange ({ operation, updatedItem, context }) {
+            if (operation === 'update' && updatedItem.type === ACCESS_RIGHT_SET_GLOBAL_TYPE) {
+                const otherTypeRightSets = await find('B2BAppAccessRightSet', {
+                    app: { id: updatedItem.app },
+                    type_not: ACCESS_RIGHT_SET_GLOBAL_TYPE,
+                    deletedAt: null,
+                })
+                for (const rightSet of otherTypeRightSets) {
+                    const diffPermissions = getPermissionsDiff(updatedItem, rightSet)
+                    const permissionsToDisable = getEnabledPermissions(diffPermissions)
+                    const disabledPermissions = Object.fromEntries(Object.keys(permissionsToDisable).map((perm) => [perm, false]))
+                    if (Object.keys(disabledPermissions).length) {
+                        await B2BAppAccessRightSetAPI.update(context, rightSet.id, {
+                            dv: rightSet.dv,
+                            sender: updatedItem.sender,
+                            ...disabledPermissions,
+                        }, 'id')
+                    }
+                }
+            }
+
+            if (operation === 'update' && updatedItem.type === ACCESS_RIGHT_SET_SCOPED_TYPE) {
+                const accessTokens = await find('B2BAccessToken', {
+                    rightSet: { id: updatedItem.id },
+                    deletedAt: null,
+                })
+                for (const accessToken of accessTokens) {
+                    await B2BAccessToken.update(context, accessToken.id, {
```

Comment:
Please add a note, why do you use empty update. I've figured out to trigger some hooks in them, but it's not obvious in the file context, so someone can delete this lines